modules/server/src/main/scala/higherkindness/compendium/transformer/skeuomorph/SkeuomorphProtocolTransformer.scala

```diff
@@ -19,31 +19,41 @@ package higherkindness.compendium.transformer.skeuomorph
 import cats.effect.Sync
 import cats.implicits._
 import higherkindness.compendium.models.{FullProtocol, IdlName, Protocol, ProtocolMetadata}
+import higherkindness.compendium.models.transformer.types._
 import higherkindness.compendium.transformer.ProtocolTransformer
 import higherkindness.skeuomorph.protobuf.ParseProto.ProtoSource
 import higherkindness.skeuomorph.{avro, mu, protobuf}
 import org.apache.avro.{Protocol => AvroProtocol}
 import higherkindness.droste.data.Mu
 import higherkindness.droste.data.Mu._
+import scala.meta._
 
 object SkeuomorphProtocolTransformer {
 
   def apply[F[_]: Sync]: ProtocolTransformer[F] = new ProtocolTransformer[F] {
 
     import higherkindness.compendium.transformer.protobuf.parsing._
 
-    private def protobufToMu(source: ProtoSource, oldMetadata: ProtocolMetadata): F[FullProtocol] =
+    private def protobufToMu(
+        source: ProtoSource,
+        oldMetadata: ProtocolMetadata,
+        streamCtor: (Type, Type) => Type.Apply
+    ): F[FullProtocol] =
       protobuf.ParseProto
         .parseProto[F, Mu[protobuf.ProtobufF]]
         .parse(source)
-        .map { protobuf =>
-          val muProto        = mu.Protocol.fromProtobufProto(mu.CompressionType.Identity, true)(protobuf)
-          val targetMetadata = oldMetadata.copy(idlName = IdlName.Mu)
-          val targetProto    = Protocol(mu.print.proto.print(muProto))
-          FullProtocol(targetMetadata, targetProto)
-        }
+        .map(protobuf => mu.Protocol.fromProtobufProto(mu.CompressionType.Identity, true)(protobuf))
+        .flatMap(p =>
+          Sync[F]
+            .fromEither(mu.codegen.protocol(p, streamCtor).leftMap(TransformError).map(_.syntax))
+        )
+        .map(p => FullProtocol(oldMetadata.copy(idlName = IdlName.Mu), Protocol(p)))
+
+    private def skeuomorphTransformation(fp: FullProtocol, target: IdlName): F[FullProtocol] = {
+      val streamCtor: (Type, Type) => Type.Apply = {
+        case (f, a) => t"_root_.fs2.Stream[$f, $a]"
```

Comment:
This will hardcode the streaming implementation to FS2, which I guess is OK for now? I'm not sure what the correct behaviour is here. In the Mu sbt plugin we allow the user to choose between FS2 and Monix using an sbt setting.

Reply:
Maybe we could provide that configuration option in future releases.